src/Mollie/WC/Plugin.php

```diff
@@ -276,11 +276,36 @@ function ($settings) {
 		self::schedulePendingPaymentOrdersExpirationCheck();
         self::registerFrontendScripts();
         wp_enqueue_style('mollie-gateway-icons');
+        self::maybeFixSubscriptions();
 
 		// Mark plugin initiated
 		self::$initiated = true;
     }
 
+    /**
+     * See MOL-322
+     */
+    public static function maybeFixSubscriptions(){
```

Comment:
I'd suggest moving this to a separate class. This way, you'd avoid polluting the `Plugin` class and get the ability to easily remove it someday when/if you decide you don't need this anymore.